src/uct/base/uct_mem.c

```diff
@@ -129,6 +130,33 @@ ucs_status_t uct_mem_alloc(void *addr, size_t min_length, unsigned flags,
             }
             break;
 
+        case UCT_ALLOC_METHOD_THP:
+            if (!ucs_is_thp_enabled()) {
+                break;
+            }
+
+            alloc_length = ucs_align_up(min_length, ucs_get_huge_page_size());
+            address = ucs_memalign(ucs_get_huge_page_size(), alloc_length
+                                   UCS_MEMTRACK_VAL);
+            if (address != NULL) {
+#ifdef MADV_HUGEPAGE
+                status = madvise(address, alloc_length, MADV_HUGEPAGE);
+                if (status != UCS_OK) {
+                    ucs_error("madvise failure status (%d) errno(%s)",
+                              status, strerror(errno));
+                    ucs_free(address);
+                } else {
+                    goto allocated_without_md;
+                }
+#else
```

Comment:
so if MADV_HUGEPAGE is not define, the memory will be allocated and released immediately? might just skip THP in this case. Or check if "always" is selected, and hope the 2MB alignment would be enough to enable THP